### DISCUSSION TITLE
Charge from Solcast forecast not working with Energy management switches? (original integration)

### DISCUSSION THREAD

**Discussion by kme77** [nathanmarlor/foxess_modbus#454] [Q&A]
Hi, I had this working last winter but turned the management off over the summer as I was generating excess energy. Now when I try to turn it back on the only options that appear to work are 'off' (no overnight charging) or full charge. Can someone please explain what the 'Disable Auto Charge' as it would seem to be obvious but does it just disable the management? (the same as turning off the main switch)
When I turn on the main switch, both 'FULL CHARGE' and 'DISABLE AUTO CHARGE' activate, I can disable FC and leave DAC active but if I switch both to off then everything is deactivated including the main switch.

I just want to get forecast charging running again so it only charges enough to get through the following day but I'm not sure what has changed since last winter.
I've attached photos of the 3 switch combinations that work and the last one which (I think) I WANT to work but won't stick when selected.
ANY help would be MUCH appreciated!

FYI, my Solcast API is working fine, I'm on HAOS 10.5, core 2023.9.3, v1.8.0 of the original (fox cloud) integration and connected via LAN to the Fox cloud. (I haven't got around to upgrading to local control using RS485 yet as I haven't got my head around the software changes required- I only got into HA when I installed my solar so it's still very confusing and don't want to break/lose anything!)
![all off](https://github.com/nathanmarlor/foxess_modbus/assets/38857063/8282da23-a315-4956-a925-f06dd57988cf)
![all on](https://github.com/nathanmarlor/foxess_modbus/assets/38857063/0f6ae984-d497-42b2-b868-cfee27b6e6fe)
![disable on](https://github.com/nathanmarlor/foxess_modbus/assets/38857063/8bdc3932-a372-4be2-8f43-e30fd812f68e)
![want but can't get](https://github.com/nathanmarlor/foxess_modbus/assets/38857063/9b56f731-2727-46f0-9cc3-024ccbe9e8fd)

**Comment by nathanmarlor** [comment]
@kme77 the 'big' toggle across the top is a red herring - all it does is act as an 'all on' or 'all off' switch for EM. Turning on 'disable auto charge' will do what you're looking for.

**Reply by kme77** [reply]
Thanks that's great, I'll try it out and keep an eye on it. 👍
Do you think a rename would be helpful, maybe 'enable forecast/solcast charge' instead? Just to make the function clearer. Thanks again, much appreciated!

**Reply by kme77** [reply]
Unfortunately it looks like this didn't work. I have 'disable auto charge' toggled on and my battery got down to MinSoc (10%) at 10pm last night, with a charge needed of 0.5kW according to FoxessEM. During the night the battery stayed discharged with the overnight load drawing from the grid and no charging occurring during the eco period. The battery even went down to 9% as I write this.
Any ideas what is happening? Btw I upgraded to your integration from Stealth Chestnut, following the recommended instructions.

Cheers.

**Reply by canton7** [reply]
Btw, you're talking about the Foxess EM integration, but you're posting in the foxess_modbus project. https://github.com/nathanmarlor/foxess_em

**Reply by kme77** [reply]
Yeah I realised that after I posted but as Nathan replied here I wanted to keep the conversation in one place. Do you know if it's possible to migrate the discussion?

**Reply by nathanmarlor** [reply]
@kme77 if you had 'disable auto charge' switched on then no charging would have occurred. As @canton7 said, let's move the discussion across to EM please.

**Reply by kme77** [reply]
How do I transfer the discussion? When I try the 'transfer this discussion' link on the right and search for your EM project nothing shows up?

**Reply by kme77** [reply]
![Capture2](https://github.com/nathanmarlor/foxess_modbus/assets/38857063/58a639d7-adc3-438f-abbd-b04e6d09130d)

**Reply by kme77** [reply]
Started a new discussion in the correct place.

**Reply by jamesog** [reply]
You should just use the repo name, `nathanmarlor/foxess_em`, not the full URL.

**Reply by kme77** [reply]
Tried that, and a bunch of variations but it didn't work either. :/ 
Hopefully somebody will offer an answer on the new thread.
